### DISCUSSION TITLE
Need help with defaultKcProps during upgrade to version 8

### DISCUSSION THREAD

**Discussion by anand-kulk** [keycloakify/keycloakify#419] [Q&A] [answered]
@garronej or anyone-
I am trying to upgrade keycloakify from version 6.8.13 to latest 8.0.0. After upgrade, I am getting errors resolving below lines in my index.js file.
```
import { defaultKcProps, getKcContext } from 'keycloakify'; 
```
it can't find defaultKcProps in path keycloakify. I tried to check under node_modules but don't see that exported anymore...I saw there is TemplateProps and PageProps types but not sure how they can replace defaultKcProps, this is how it is being used in index.js -

```
import React from 'react';
import ReactDOM from 'react-dom/client';
import { KcApp } from './components/KcApp/';
import { defaultKcProps, getKcContext } from 'keycloakify';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { ForgotPassword, Login, UpdatePassword } from './pages';
import './App.scss';
import { Footer } from '../src/components/shared/Footer';

const root = ReactDOM.createRoot(document.getElementById('root'));

const { kcContext } = getKcContext();

const app = (
  <React.StrictMode>
    <div
      id='mainContainer'
      className='flex flex-col h-screen bg-surfaceBackgroundLight'
    >
      <Router>
        <Routes>
          <Route
            exact
            path='/forgot-password'
            element={
              <ForgotPassword
                kcContext={kcContext}
                {...{
                  ...defaultKcProps,
                }}
              />
            }
          />
```

Any help will be appreciated.

Anand

**Comment by garronej** [comment]
Hello,  
Sorry about hearing you have difficulties to update.  
Have you checked the migration guides?  

https://docs.keycloakify.dev/migration-guides

**Reply by anand-kulk** [reply]
I had taken a look at them but didn't see anything specific said for `defaultKcProps`

**Reply by garronej** [reply, marked as answer]
In defaultKcProps there was the default classes https://github.com/keycloakify/keycloakify/blob/9e75ee09bb898678156ebcf37134f56bf5de739c/src/lib/KcProps.ts#L134-L209. 

Now we use getClassNames instead https://github.com/keycloakify/keycloakify-starter/blob/92b20fe74154ef8cf037f4b156eb3b2e5264a074/src/keycloak-theme/login/pages/Login.tsx#L12-L15. 

Hope it helps

**Reply by anand-kulk** [reply]
Thanks for clarifying, I'll try getClassNames. Appreciate your help.
